### DISCUSSION TITLE
Breadcrumb Navigation

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#224] [Component suggestions]
<img width="480" alt="visuele weergave van het breadcrumb-navigation component" src="https://github.com/nl-design-system/backlog/assets/545207/a043e97b-1eac-464b-a81d-f4c7a93d789e">

We willen graag meer leren over het **Breadcrumb Navigation** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1708359932709109)

---

## Naam
Breadcrumb Navigation

## Doel
Toont de locatie van de pagina binnen de hiërarchie van een website en biedt de mogelijkheid om tussen niveaus te navigeren.

## Onderzoek

[Breadcrumbs: 11 Design Guidelines for Desktop and Mobile - Nielsen Norman Group](https://www.nngroup.com/articles/breadcrumbs/)
Artikel over Breadcrumb Navigation met richtlijnen. Onder andere over [home](https://www.nngroup.com/articles/breadcrumbs/#:~:text=start%20with%20a-,link%20to%20the%20homepage,-.), [current](https://www.nngroup.com/articles/breadcrumbs/#:~:text=Include%20the-,current%20page,-as%20the%20last) en [responsive gedrag](https://www.nngroup.com/articles/breadcrumbs/#:~:text=use%20breadcrumbs%20that-,wrap%20to%20multiple%20lines,-.).

[Designing Effective Breadcrumbs Navigation - Smashing Magazine](https://www.smashingmagazine.com/2022/04/breadcrumbs-ux-design/)
Artikel over Breadcrumb Navigation met richtlijnen. Onder andere over de [positie](https://www.smashingmagazine.com/2022/04/breadcrumbs-ux-design/#:~:text=If%20at%20any%20point%20we%20need%20to%20move%20backwards%20or%20sideways%2C%20we%20need%20to%20be%20able%20to%20find%20breadcrumbs%20immediately.%20Displaying%20breadcrumbs%20just%20under%20the%20main%20navigation%20or%20just%20above%20the%20main%20heading%20is%20the%20most%20obvious%20and%20familiar%20way%20to%20achieve%20just%20that.), home, [underline](https://www.smashingmagazine.com/2022/04/breadcrumbs-ux-design/#that-question-about-that-last-item:~:text=Plain%20underlines%20would%20make%20it%20slightly%20more%20obvious), [current](https://www.smashingmagazine.com/2022/04/breadcrumbs-ux-design/#that-question-about-that-last-item) en [responsive gedrag](https://www.smashingmagazine.com/2022/04/breadcrumbs-ux-design/#that-question-about-that-last-item:~:text=wrapped%20into%20multiple%20lines).

[6 Important Aspects of Well-Performing Mobile Product Page Breadcrumbs - Baymard Institute](https://baymard.com/blog/implementing-mobile-hierarchy-breadcrumbs)
Artikel over Breadcrumb Navigation met richtlijnen. Onder andere over responsive gedrag.

In een [comment bij Breadcrumb Github Discussion van GOV.UK](https://github.com/alphagov/govuk-design-system-backlog/issues/33#issuecomment-903872698) wordt opgemerkt dat in een onderzoek niet op de Breadcrumb werd geklikt omdat deze dezelfde kleur had als de tekst.

In een [Github Issue bij WCAG Audit Discussion](https://github.com/WCAG-Audit-Discussions/NL-BE/issues/59) wordt besproken hoe om te gaan met een Breadcrumb bij een small viewport (dan wel ingezoomd of lage resolutie).

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-navigation-breadcrumb--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/breadcrumb)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=7744-95754&mode=design&t=mWmi5RlnWfGtwSpk-4)

---

### Screenshots / Varianten

**Storybook**

Large viewport
![Scherm­afbeelding 2024-03-06 om 13 15 19](https://github.com/nl-design-system/backlog/assets/545207/ed59da27-fa1b-4ae6-bf80-e7f7b3a598a9)

---

### Notities
- Home: Uitgeschreven als link.
- Current page: Lijkt geen deel uit te maken van breadcrumb.
- Small viewport: Over meerdere regels.
- Richtlijnen beschikbaar (Storybook).

**Comment by sanderdegoose** [comment]
## Rotterdam

- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=501-18760&mode=design&t=X9RCEbKDEdvg0izr-4)
- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/css-utrecht-breadcrumb-nav--docs)

---

### Screenshots / Varianten

**Figma**

Default
![Scherm­afbeelding 2024-03-06 om 16 04 26](https://github.com/nl-design-system/backlog/assets/545207/3ecb9ef4-cb95-4d12-b7a0-19c56daf33e9)

Truncated
![Scherm­afbeelding 2024-03-06 om 16 04 35](https://github.com/nl-design-system/backlog/assets/545207/6402f1ef-b42b-4278-9c54-61e5d1cceacb)

Small viewport
![Scherm­afbeelding 2024-03-06 om 16 05 58](https://github.com/nl-design-system/backlog/assets/545207/8eb7847e-1214-4349-841d-eab4fbc4b8e1)


**Storybook**

![Scherm­afbeelding 2024-03-06 om 16 08 14](https://github.com/nl-design-system/backlog/assets/545207/789b8674-5ceb-4ded-bf95-fda201e1c570)

---

### Notities
- Home: Uitgeschreven als link.
- Current page: Maakt als laatste item deel uit van breadcrumb (kleur wordt in Storybook nog aangepast).
- Small viewport:
  - In Storybook over meerdere regels.
  - In Figma enkel een link naar het bovenliggende niveau.
  - Op de [rotterdam.nl](https://www.rotterdam.nl/gemeenteraad) wordt 'Home' weggelaten.
- Richtlijn (Figma) beschrijft dat breadcrumb link bij een maximaal aantal karakters truncated kan worden.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-breadcrumb--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Breadcrumb)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1569-5671&mode=design&t=5cajoGp4SRLyk87x-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=1862-9576&mode=design&t=5yAEsnjVXyQwuybQ-4)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-03-06 om 13 23 13](https://github.com/nl-design-system/backlog/assets/545207/870def32-c509-412c-b319-c184596d89ea)

Long
![Scherm­afbeelding 2024-03-06 om 13 23 22](https://github.com/nl-design-system/backlog/assets/545207/c72887d9-9d59-4d34-b295-f97e4a032dbc)

With icon
![Scherm­afbeelding 2024-03-06 om 13 23 29](https://github.com/nl-design-system/backlog/assets/545207/796372be-b3f6-4408-8a46-ad743945b2d7)

Without current
![Scherm­afbeelding 2024-03-06 om 13 23 37](https://github.com/nl-design-system/backlog/assets/545207/a21a57a6-5863-4553-bf56-b7a30aa4b297)

---

### Notities
- Home: Uitgeschreven als link. Optie beschikbaar om als icoon (huis) te tonen.
- Current page: Maakt als laatste item deel uit van breadcrumb. Optie beschikbaar niet te tonen.
- Small viewport: Enkel een link naar het bovenliggende niveau.
- Truncation als optie bij lange breadcrumb.
- Richtlijnen beschikbaar.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

[Documentatie website](https://dso-toolkit.nl/62.8.4/components/navigatie/breadcrumbs)

---

### Screenshots / Varianten

**Documentatie website**
![Scherm­afbeelding 2024-03-06 om 14 01 33](https://github.com/nl-design-system/backlog/assets/545207/d9f6ee06-720b-4b93-92b9-a9f88f48200a)

---

### Notities
- Home: Uitgeschreven als link.
- Current page: Maakt als laatste item deel uit van breadcrumb.
- Small viewport: Over meerdere regels.
- Richtlijnen beschikbaar.

**Comment by jeffreylauwers** [comment]
## DUO

- [Documentatie website - Backlink](https://uno.dfront.rijkscloud.nl/#/components/backlink)
- [Documentatie website - Navgation](https://uno.dfront.rijkscloud.nl/#/components/navigation)

---

### Screenshots / Varianten

![Scherm­afbeelding 2024-03-06 om 14 17 30](https://github.com/nl-design-system/backlog/assets/545207/21e6e40d-3785-4f75-aa8d-fbce44a5ab8d)

---

### Notities
Geen specifiek Breadcrumb navigation component. Er wordt gebruik gemaakt van een mix aan componenten (navigation, back-link en list-links). Het Navigation component biedt een vorm van breadcrumb-ish gedrag.

**Comment by jeffreylauwers** [comment]
## OpenGemeenten
- [Patternlab](https://patternlab.lenteveld.nl/index.html)
- [Voorbeeld website](https://www.lenteveld.nl/bestuur-en-organisatie/nieuwbouw-in-lenteveld)

---

### Screenshots / Varianten

**Patternlab**

[Bread Crumb](https://patternlab.lenteveld.nl/patterns/01-molecules-001-Generic-Page-Components-30-Main-Top-01-OpenGemeentenSite-BreadCrumb-OpenGemeentenSite-BreadCrumb/01-molecules-001-Generic-Page-Components-30-Main-Top-01-OpenGemeentenSite-BreadCrumb-OpenGemeentenSite-BreadCrumb.html)

![Scherm­afbeelding 2024-03-06 om 14 23 23](https://github.com/nl-design-system/backlog/assets/545207/6c7e4b9d-de97-4832-aaf8-a4f6da8193a3)

---

### Notities
- Home: Als icoon (huis).
- Current page: Maakt als laatste item deel uit van breadcrumb.
- Small viewport: Icoon (arrow-left) en vanaf bepaald breakpoint over meerdere regels.
- Momenteel wordt er bij Open Gemeenten gekeken naar het gedrag van de Breadcrumb navigation bij een small viewport.

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-breadcrumbs--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/breadcrumbs)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4207-18664&mode=design&t=5PJ1XiTbtHkHlQum-4)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-03-06 om 16 58 06](https://github.com/nl-design-system/backlog/assets/545207/2782f083-1c91-4739-b45e-fb6766625139)

---

### Notities
- Home: Uitgeschreven als link ([zie rvo.nl](https://www.rvo.nl/onderwerpen/financiering-ondernemers-fase/starten)).
- Current page: Maakt als laatste item deel uit van breadcrumb.
- Small viewport: Over meerdere regels ([zie rvo.nl](https://www.rvo.nl/onderwerpen/financiering-ondernemers-fase/starten)).

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-breadcrumb-nav--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/breadcrumb-nav)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=1862-9576&mode=design&t=LVnyg6GmH5PCgjBf-4)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-03-06 om 17 07 38](https://github.com/nl-design-system/backlog/assets/545207/bb0347b7-1901-4c10-ab90-3e42334ba366)

---

### Notities
- Home: Uitgeschreven als link.
- Current page: Maakt als laatste item deel uit van breadcrumb (zelfde kleur als interactieve items).
- Small viewport: In Storybook over meerdere regels. In Figma enkel een link naar het bovenliggende niveau.
- Verschillende opties beschikbaar (Storybook).
- Documentatie beschikbaar (Storybook).

**Reply by Rozerinay** [reply]
Opmerking afkomstig uit een comment in de NL Design System Figma bibliotheek:

Token `utrecht.breadcrumb-nav.text-transform` is niet opgenomen in het voorbeeldthema (JSON) in Figma, omdat uit overleg over Breadcrumb Navigation blijkt dat dit type teksttransform in de praktijk niet wordt gebruikt.

**Reply by Rozerinay** [reply]
## Toelichting op opnemen van ‘current’ pagina

Er bestaat verschil in implementatie tussen organisaties over het opnemen van de huidige pagina als laatste item in het kruimelpad. In de Utrecht README staat dat de pagina zelf geen onderdeel is van het kruimelpad, terwijl het codevoorbeeld datzelfde item wel toont. Andere organisaties nemen de current pagina juist wél op.

Na overleg is besloten: als de huidige pagina duidelijk te herleiden is uit de paginatitel en Heading 1, kan het laatste item in de Breadcrumb Navigation worden weggelaten. Tegelijk leeft het idee dat het tonen van de current pagina bijdraagt aan het begrip van de structuur. Voor dit laatste is aanvullend gebruikersonderzoek nodig.

Rozerin verwijst naar aanbevelingen van o.a. Vitaly Friedman en NNGroup:

- Als Breadcrumb Navigations direct boven de Heading 1 staan, mag de huidige pagina weggelaten worden mits semantiek en toegankelijkheid goed zijn geregeld.
- Hoe verder de Breadcrumb Navigation visueel van de Heading 1 afstaat, hoe belangrijker het wordt om de huidige pagina wél te tonen in de trail.

Bronnen:
- [NNGroup – Breadcrumbs](https://www.nngroup.com/articles/breadcrumbs/#:~:text=Include%20the%20current%20page%20as%20the%20last%20item%20in%20the%20breadcrumb%20trail)
- [Smashing Magazine – Breadcrumb UX Design](https://www.smashingmagazine.com/2022/04/breadcrumbs-ux-design/#that-question-about-that-last-item:~:text=That%20Last%20Item)

Bron: uitwisseling tussen @Rozerinay en @jeffreylauwers in de NL Design System Figma bibliotheek.

**Reply by Rozerinay** [reply]
## Toelichting op `todo.breadcrumb-nav.link.text-decoration`

De token `todo.breadcrumb-nav.link.text-decoration` kan worden ingezet om Breadcrumb links te onderscheiden van reguliere links, bijvoorbeeld door het ontbreken van een underline. Dit helpt om de current link niet alleen via kleur, maar ook via vorm visueel te onderscheiden.

Bron: opmerking in de NL Design System Figma bibliotheek (2024).

**Reply by Rozerinay** [reply]
## Toelichting op `todo.breadcrumb-nav.link.active.color`

Voor `todo.breadcrumb-nav.link.active.color` is het de vraag of een aparte token nodig is, of dat de styling van de active state overgenomen kan worden vanuit de Link component.

Bron: opmerking in de NL Design System Figma bibliotheek (2024).

**Reply by Rozerinay** [reply]
## Toelichting op `todo.breadcrumb-nav.link.current.color`

In de code is voor de current link al een `font-weight` beschikbaar. Een aanvullende token voor color, zoals `todo.breadcrumb-nav.link.current.color`, zou echter ook waardevol kunnen zijn voor extra visueel onderscheid.

Bron: opmerking in de NL Design System Figma bibliotheek (2024).

**Reply by Rozerinay** [reply]
## Toelichting op `todo.breadcrumb-nav.link.icon.size` en `.column-gap`

De tokens `todo.breadcrumb-nav.link.icon.size` en `todo.breadcrumb-nav.link.icon.column-gap` zouden mogelijk kunnen worden geërfd vanuit de algemene Link component, in plaats van specifiek voor Breadcrumb Navigation te worden gedefinieerd.

Bron: opmerking in de NL Design System Figma bibliotheek (2024).

**Comment by jeffreylauwers** [comment]
Nog een variant voor Breadcrumb navigation. Gedeeld door Niels Roozemond tijdens een Design Open Hour:
https://gui-challenges.web.app/breadcrumbs/dist/

**Comment by jeffreylauwers** [comment]
## Belastingdienst

[Detailpagina op website](https://www.belastingdienst.nl/wps/wcm/connect/nl/belastingaangifte/content/problemen-inloggen-gebruik-mijn-belastingdienst)

---

### Screenshots / Varianten

Large viewport
![Scherm­afbeelding 2024-04-30 om 16 25 00](https://github.com/nl-design-system/backlog/assets/545207/46af49d4-e91c-4ce5-93a0-ba45527625b9)

Small viewport
![Scherm­afbeelding 2024-04-30 om 16 25 17](https://github.com/nl-design-system/backlog/assets/545207/cd71a4ab-2158-4979-9ca6-c08f399699f4)

---

### Notities
Small viewport: Breadcrumb wordt in zijn geheel weggelaten waarschijnlijk onder het motto 'Leuker kunnen we het niet maken, wel makkelijker' 😉

**Comment by Robbert** [comment]
Alle voorbeelden van Breadcrumb navigation zijn onderdeel van de navigatie van de pagina, en zijn daarom waarschijnlijk onderdeel van een `navigation` landmark.

Er is ook nog een andere toepassing van breadcrumb navigation, daar moet component niet onderdeel zijn van de `navigation` landmark: de breadcrumb navigation in zoekresultaten. Ik kan me niet herinneren dat ik deze versie recent tegen ben gekomen op een overheidswebsite.

![breadcrumb-in-search-results](https://github.com/nl-design-system/backlog/assets/30694/ca8fc85f-4c0a-437f-851a-93dd3e5b64d6)

**Reply by Robbert** [reply]
Om dit in code mogelijk te maken, zou het misschien een idee zijn om de implementatie op te splitsen in een `<BreadcrumbNavigation>` voor de navigation landmark versie, en een `<BreadrcrumbLinkList>` voor de zoekresultatenversie. De `<BreadcrumbNavigation>` kan dan intern gebruik maken van de `<BreadcrumbLinkList>`.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Breadcrumb navigation op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=breadcrumb+navigation)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Breadcrumb navigation op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=component%3A+breadcrumb+navigation)

**Comment by jeffreylauwers** [comment]
## MijnOverheid

Bij het bekijken van een bericht binnen MijnOverheid kwam ik een Breadcrumb tegen met responsive gedrag waarbij enkel het bovenliggende niveau wordt getoond bij een smallviewport (in dit geval portrait-mode).

Portrait-mode
<img width="500" height="450" alt="portrait" src="https://github.com/user-attachments/assets/30c3725c-6155-499d-8170-3a078de36d3c" />

---

Landscape-mode
<img width="2126" height="839" alt="landscape" src="https://github.com/user-attachments/assets/a8bc40ef-2dfb-4279-a667-e7b3738becc1" />
